Help page document

georges1996 opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

Handle Help page documents in Frontend, in preparation for them being taken out of Government Frontend

## Why

https://trello.com/c/GP0bkcD4/350-move-document-type-helppage-from-government-frontend-to-frontend, [Jira issue PNP-8507](https://gov-uk.atlassian.net/browse/PNP-8507)

## How

These documents all have a fixed prefix, so add a scoped route for them, a controller, a model and a view. Here we add the initial ContentItem model the document type models will be based on, with the minimum functionality required to get Help page documents working.

We also create a couple of helpers for rendering generic content.

## Screenshots?

From example page: https://www.gov.uk/help/browsers

### Before (running in Government Frontend)
<img width="749" alt="Screenshot 2024-10-04 at 07 54 53" src="https://github.com/user-attachments/assets/616f54f6-f61e-415d-806a-879e1b23c939">


### After (running in Frontend)
<img width="631" alt="Screenshot 2024-10-04 at 07 54 35" src="https://github.com/user-attachments/assets/a0b82ac3-6915-4fba-b8eb-f91c3fe2a8fb">

georges1996 committed:
Update README.md

georges1996 committed:
Add footer and sidebar shared partials

Commit audit trail:
- app/views/shared/_footer_navigation.html.erb https://github.com/alphagov/government-frontend/blob/3e5710c5129b3e1b1abd7176fef951bc73b7b49b/app/views/shared/_footer_navigation.html.erb
- app/views/shared/_sidebar_navigation.html.erb https://github.com/alphagov/government-frontend/blob/3e5710c5129b3e1b1abd7176fef951bc73b7b49b/app/views/shared/_sidebar_navigation.html.erb

georges1996 committed:
Add help_page controller and update content item model to support help_page

- Add new route
- Add help_page controller
- Add body_with_related_links partial
- Update content item model
- Add help_page views
- Add system  spec for the new help_page

Commit audit trail:
- spec/system/help_page_spec.rb https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/test/integration/help_page_test.rb
- app/views/help_page/show.html.erb https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/views/content_items/help_page.html.erb
- app/views/content_items/_body_with_related_links.html.erb https://github.com/alphagov/government-frontend/blob/3e5710c5129b3e1b1abd7176fef951bc73b7b49b/app/views/content_items/_body_with_related_links.html.erb

georges1996 committed:
Add DateHelper and add new method in LocaleHelper

- Add new DateHelper to deal with the last updated date on help pages
- Add page_text_direction to LocaleHelper

Commit audit trail:
- app/helpers/locale_helper.rb https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/helpers/application_helper.rb
- app/helpers/date_helper.rb https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/presenters/content_item_presenter.rb

leenagupte commented on spec/helpers/date_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+RSpec.describe DateHelper do
```
Nice tests 🎉

leenagupte commented on app/views/help_page/show.html.erb:
```diff
@@ -0,0 +1,9 @@
+<% content_for :extra_head_content do %>
```
This should be `extra_headers` not `extra_head_content`. It's something that tripped up both myself and Keith.

You can check if the metadata is being displayed by viewing the page source. If the metaproperty appears it's all ok:

```
<meta property="og:description" content="How to upgrade your web browser - improve safety, speed and features">
<meta property="og:title" content="Browsers">
<meta property="og:url" content="https://www.gov.uk/help/browsers">
<meta property="og:type" content="article">
<meta property="og:site_name" content="GOV.UK">
<meta name="twitter:card" content="summary">
```
To combat this Keith started porting over the [meta tags tests](https://github.com/alphagov/frontend/pull/4218/commits/975dca796779ef0a3ec0b5c3d836488164371e3e) as shared examples that could be added to all of the system tests. Perhaps that could be done in this PR instead?

leenagupte reviewed (commented):
This is a really good start. I think not being able to merge the Take Part PR means that this one needs some extra work before it's ready.

We've already spoken IRL about linking to the files in government-frontend directly. I think the only commit that needs the links updated is the one that ports over the components.

I've added inline comments for everything else.

georges1996 committed:
Add meta tags spec support and add to HelpPage spec

- test/integration/meta_tags_test.rb https://github.com/alphagov/government-frontend/blob/5d6622e384dcde9c19fcefe88d94414e1d6fb457/test/integration/meta_tags_test.rb

KludgeKML commented on app/models/content_item.rb:
```diff
@@ -10,6 +10,8 @@ def initialize(content_store_response)
     @title = content_store_response["title"]
     @base_path = content_store_response["base_path"]
     @locale = content_store_response["locale"]
+    @last_updated = content_store_response["public_updated_at"]
```
Hmm. I sort of think here we should be keeping the attributes named the same as they are in the content store - like, this should be `public_updated_at`

georges1996 committed:
Add Published Dates component
- Used in Help Page
- Includes spec file, documentation

Commit audit trail:
- app/assets/stylesheets/components/_published-dates.scss https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/assets/stylesheets/components/_published-dates.scss
- app/views/components/_published_dates.html.erb https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/views/components/_published_dates.html.erb
- app/views/components/docs/published_dates.yml https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/app/views/components/docs/published_dates.yml
- spec/components/published_dates_spec.rb https://github.com/alphagov/government-frontend/blob/af54031948650448b965ac274c3248d23658c4f6/test/components/published_dates_test.rb

georges1996 committed:
Add missing translation files for published date components
-  Ran rake "consolidation:copy_translation[components.published_dates]"

georges1996 committed:
Add translation files for direction

- Ran rake "rake "consolidation:copy_translation[i18n]"

KludgeKML reviewed (approved):
LGTM (but we need a second opinion because I am now implicated in some of this)

leenagupte reviewed (commented):
Looks good! 

There are a couple of inline comments about linking to the original files in government-frontend, otherwise it looks great. 🎉

leenagupte reviewed (approved):
Looks great 👍
